### PR TITLE
Split PR Part 5: Extended Environment type. New auth, new matchers

### PR DESCRIPTION
**Pull request by stopthatastronaut**
Does the thing it says up there

### CODE REVIEW

File: docs/octopus_deploy_environment.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# octopus_deploy_environment
+
+Models/Tests an Octopus Deploy [Environment](https://octopus.com/docs/infrastructure/environments) resource
```
Comment (matt-richardson) [inline docs/octopus_deploy_environment.md]:
```suggestion
Models/Tests an Octopus Deploy [Environment](https://octopus.com/docs/infrastructure/environments) resource.
```

Is this the same wording you've used elsewhere?

File: docs/octopus_deploy_environment.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+```ruby
+describe octopus_deploy_environment('Production').in_space('Team Octopus') do
+  it { should exist }
+end
+```
+
+#### Type
+
+This type can be instantiated in several ways, depending on [how you authenticate](authentication.md), and whether your target environment is in a [Space](https://octopus.com/blog/spaces-introduction)
```
Comment (matt-richardson) [inline docs/octopus_deploy_environment.md]:
```suggestion
This type can be instantiated in several ways, depending on [how you authenticate](authentication.md), and whether your target environment is in a [Space](https://octopus.com/blog/spaces-introduction).
```

File: lib/octopus_serverspec_extensions/matcher/use_guided_failure.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+RSpec::Matchers.define :use_guided_failure do
```
Comment (matt-richardson) [inline lib/octopus_serverspec_extensions/matcher/use_guided_failure.rb]: what do these matchers do that the type doesn't?

Comment (stopthatastronaut) [reply lib/octopus_serverspec_extensions/matcher/use_guided_failure.rb]: Let's see if I can word this correctly... There are some common matcher verbs that spec automatically 'translates' (for want of a better word) into a method call. A method on the type ends up looking a bit clumsy. example:

`should be_in_space -calls-> .in_space?`
`.has_description? -> expect(thing).to have_description`

It won't do that kind of thing with "use" unless you go down the matcher route. Which gives you both worlds.

File: docs/octopus_deploy_environment.md (hunk 1 above)
Comment (stopthatastronaut) [reply docs/octopus_deploy_environment.md]: It really isn't. I'll fix that

**Pull request merged**